tests/tensor/test_nlinalg.py

```diff
@@ -580,3 +581,42 @@ def test_eval(self):
         t_binv1 = tf_b1(self.b1)
         assert _allclose(t_binv, n_binv)
         assert _allclose(t_binv1, n_binv1)
+
+
+class TestKron(utt.InferShapeTester):
+    rng = np.random.default_rng(43)
+
+    def setup_method(self):
+        self.op = kron
+        super().setup_method()
+
+    def test_perform(self):
+        for shp0 in [(2,), (2, 3), (2, 3, 4), (2, 3, 4, 5)]:
+            x = tensor(dtype="floatX", shape=(None,) * len(shp0))
+            a = np.asarray(self.rng.random(shp0)).astype(config.floatX)
+            for shp1 in [(6,), (6, 7), (6, 7, 8), (6, 7, 8, 9)]:
+                if len(shp0) + len(shp1) == 2:
+                    continue
+                y = tensor(dtype="floatX", shape=(None,) * len(shp1))
+                f = function([x, y], kron(x, y))
+                b = self.rng.random(shp1).astype(config.floatX)
+                out = f(a, b)
+                # Using the np.kron to compare outputs
+                np_val = np.kron(a, b)
+                np.testing.assert_allclose(out, np_val)
+
+    def test_kron_commutes_with_inv(self):
+        for shp0 in [(2, 3), (2, 4, 3)]:
+            x = tensor(dtype="floatX", shape=(None,) * len(shp0))
+            a = np.asarray(self.rng.random(shp0)).astype(config.floatX)
+            for shp1 in [(6, 7), (4, 3, 5)]:
+                if len(shp0) != len(shp1):
+                    continue
+                if len(shp0) + len(shp1) == 2:
+                    continue
+                y = tensor(dtype="floatX", shape=(None,) * len(shp1))
+                f = function([x, y], kron(x, y))
+                b = self.rng.random(shp1).astype(config.floatX)
+                lhs = matrix_inverse(f(a, b))
```

Comment:
problem is here I think. `lhs` and `rhs` are going to be symbolic variables, because `matrix_inverse` expects a pytensor symbol as input. I suggest to compile two symbolic functions:

```
f1 = function([x, y], matrix_inverse(kron(x, y)))
f2 = function([x, y], kron(matrix_inverse(x), matrix_inverse(y))
```
Then use `np.testing.assert_allclose(f1(a, b), f2(a, b))`

Reply:
oh yea this seems like a much better way to implement. i guess what i did earlier was sorta like "hard coded"

Reply:
Doing it you way is fine, the problem was that in the end we were giving a symbolic tensor to the numpy equality check. You could have used `np.linalg.inv` in that case.